Sources/TuistCore/Graph/XCFrameworkNode.swift

```diff
@@ -0,0 +1,65 @@
+import Basic
+import Foundation
+import TuistSupport
+
+final class XCFrameworkNode: PrecompiledNode {
+    let libraries: [XCFrameworkInfoPlist.Library]
+
+    init(path: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) {
+        self.libraries = libraries
+        super.init(path: path)
+    }
+
+    static func parse(
+        path: AbsolutePath,
+        cache: GraphLoaderCaching,
+        configurationFile: String = "Info.plist",
+        fileHandler: FileHandling = FileHandler.shared
+    ) throws -> XCFrameworkNode {
+        if let xcframeworkNode = cache.precompiledNode(path) as? XCFrameworkNode {
+            return xcframeworkNode
+        }
+        
+        let infoPlist = path.appending(component: configurationFile)
+        guard fileHandler.exists(infoPlist) else {
+            throw GraphLoadingError.missingFile(infoPlist)
+        }
+        
+        let config: XCFrameworkInfoPlist = try fileHandler.readPlistFile(infoPlist)
+        // TODO: implement library info.plist parsing
+        let xcframeworkNode = XCFrameworkNode(path: path, libraries: config.libraries)
+        cache.add(precompiledNode: xcframeworkNode)
+        return xcframeworkNode
+    }
+
+    override var binaryPath: AbsolutePath {
+        return XCFrameworkNode.binaryPath(frameworkPath: path, libraries: libraries)
+    }
+	
+    override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(path.pathString, forKey: .path)
+        try container.encode(name, forKey: .name)
+        try container.encode("precompiled", forKey: .type)
+    }
+	
+    /// Given a framework path and libraries it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    /// - Parameter libraries: Framework available libraries
+    static func binaryPath(
+        frameworkPath: AbsolutePath,
+        libraries: [XCFrameworkInfoPlist.Library]
+    ) -> AbsolutePath {
+        let library = libraries.first { $0.architectures.contains(.arm64) }!
+        let binaryName = frameworkPath.basenameWithoutExt
+        let binaryPath =  AbsolutePath(library.identifier, relativeTo: frameworkPath)
+            .appending(RelativePath(library.path.pathString))
+            .appending(component: binaryName)
+
+        // Sanitising to make sure the precompiled binary located at binaryPath
+        _ = try! PrecompiledMetadataProvider().architectures(binaryPath: binaryPath).first!
```

Comment:
Is this needed?

I believe the Graph methods https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Graph/Graph.swift#L461 will call this when doing look-up so we don't need to repeat it here

Reply:
as i am not using `frameworkUsesDynamicLinking` during lookup, therefore used the `PrecompiledMetadataProvider` directly here